backend/project_alpha/api/views.py

```diff
@@ -96,25 +96,37 @@ def get(self, request, nickname):  # pylint: disable=unused-argument, arguments-
         return Response(user_data)
 
 
-class UserPreferencesAPIView(APIView):
+class UserPreferencesAPIView(generics.GenericAPIView):
     """
     Retrieve user preferences.
     """
-    permission_classes = (IsAuthenticated,)
-    # TODO: APIView does not have serializer_class
-    # Probably need to use Generic views here
     serializer_class = UserPreferencesSerializer
+    permission_classes = (IsAuthenticated,)
 
-    def get(self, request):
-        user = request.user
+    def get(self, request):  # pylint: disable=unused-argument
+        user = self.request.user
         user_data = {
```

Comment:
I think we can write model serializer here, so we can have something like: `data = UserPerferencesSerializer(user.usersettings)`